src/main/java/org/spongepowered/api/item/inventory/custom/CustomInventoryBuilder.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.item.inventory.custom;
+
+import org.spongepowered.api.item.inventory.Inventory;
+import org.spongepowered.api.text.Text;
+import org.spongepowered.api.util.ResettableBuilder;
+
+/**
+ * A Builder for inventories that have no backing in the world that are created by plugins.
+ */
+public interface CustomInventoryBuilder<I extends Inventory, B extends CustomInventoryBuilder<I, B>> extends ResettableBuilder<I, B> {
+
+    /**
+     * Sets the title of the custom inventory, viewable by players looking
+     * at the inventory.
+     * TODO may not work with all Inventories -- investigate
+     *
+     * @param title The title
+     * @return Fluent pattern
+     */
+    B title(Text title);
```

Comment:
That's not the point here. The point is that Container titles **cannot** utilise rich text features, and so there is no sensible argument to **not** enforce them being localisable other than "I'm a lazy coder", which isn't strong enough to counteract the benefit of container titles being localisable.

This is a decision I made right back at the start of designing the inventory API, and it reflects the importance of the UI elements which are Containers. We're not talking about sending localisation keys to clients, we're talking about server-side translations. Making the title a translation key means that:

* responsible coders will use translation keys in their titles and provide default translations with their plugin, **which can then be set using the normal method by server owners if they have localisations available**
* lazy coders can still pass in non-translatable "keys"

All containers should have translatable titles, literally the only argument I can currently see for this not being the case is lazy coders.